crypto/evp/pmeth_lib.c

```diff
@@ -1528,23 +1528,24 @@ int EVP_PKEY_CTX_ctrl(EVP_PKEY_CTX *ctx, int keytype, int optype,
 {
     int ret = 0;
 
-    /* If unsupported, we don't want that reported here */
-    ERR_set_mark();
-    ret = evp_pkey_ctx_store_cached_data(ctx, keytype, optype,
-                                         cmd, NULL, p2, p1);
-    if (ret == -2) {
-        ERR_pop_to_mark();
-    } else {
-        ERR_clear_last_mark();
-        /*
-         * If there was an error, there was an error.
-         * If the operation isn't initialized yet, we also return, as
-         * the saved values will be used then anyway.
-         */
-        if (ret < 1 || ctx->operation == EVP_PKEY_OP_UNDEFINED)
-            return ret;
+    if (ctx != NULL) {
+        /* If unsupported, we don't want that reported here */
+        ERR_set_mark();
+        ret = evp_pkey_ctx_store_cached_data(ctx, keytype, optype,
+                                             cmd, NULL, p2, p1);
+        if (ret == -2) {
+            ERR_pop_to_mark();
+        } else {
+            ERR_clear_last_mark();
+            /*
+             * If there was an error, there was an error.
+             * If the operation isn't initialized yet, we also return, as
+             * the saved values will be used then anyway.
+             */
+            if (ret < 1 || ctx->operation == EVP_PKEY_OP_UNDEFINED)
+                return ret;
+        }
     }
```

Comment:
If you look at the only other caller of evp_pkey_ctx_ctrl_int() you can see that the function cannot be called with NULL ctx from there. So it would make sense instead of deferring the NULL check for ctx to the evp_pkey_ctx_ctrl_int() simply do it here and not call the internal function in that case.

Reply:
EVP_PKEY_CTX_ctrl() is a public function that does something with the ctx, calling it with NULL doesn't make any sense. If you want to add a check for ctx being NULL, do it at the start of the function and return an error in case it's NULL.

Reply:
I guess I'm saying the same as @t8m.